### PR TITLE
Update pdf library to WeasyPrint

### PR DESCRIPTION
**Pull request by reinbach**
[ch22072]

`easy_pdf` is no longer being maintained and having issues with py3
Changed to `WeasyPrint` library

### CODE REVIEW

File: django_api/django_api/apps/unicef/exports/programme_documents.py [hunk 1]
```diff
@@ -213,12 +212,14 @@ def get_context(self):
...
-            response['Content-disposition'] = 'inline; filename="{}"'.format(self.file_name)
+            response = render_pdf_to_response(
+                request,
+                self.template_name,
+                self.get_context(),
+            )
             return response
-        except PDFRenderingError:
-            logger.exception('Error trying to render PDF')
+        except Exception as exc:
```
Comment (robertavram) [inline django_api/django_api/apps/unicef/exports/programme_documents.py:223]: seems pretty generic

File: django_api/django_api/apps/unicef/templates/programme_documents_pdf_export.html [hunk 2]
```diff
@@ -7,31 +7,6 @@
...
         }
-
-        #pdf {
-            font-size: 10px;
-        }
-
-        h1 {
-            font-size: 11px;
-            font-weight: bold;
-        }
```
Comment (robertavram) [inline django_api/django_api/apps/unicef/templates/programme_documents_pdf_export.html:18]: is this css breaking something or just redundant?

**Review by robertavram — COMMENTED** [review]
there's a lot of css tweaking, was there any change in logic on how and where that gets applied?

File: django_api/django_api/apps/unicef/templates/programme_documents_pdf_export.html (hunk 2 above)
Comment (reinbach) [reply django_api/django_api/apps/unicef/templates/programme_documents_pdf_export.html:18]: WeasyPrint requires you to make use of `FontConfiguration` if you make use of `@font-face` styling. Which we have in the `custom_pdf_base` file. So I pulled out most of the css and placed it in a separate file.
Had to leave in the `@page` styling in the html template, as that only appeared to work there. 

The styling tweaks I did was to add;
```
table {
            width: 100%;
            border-collapse: collapse;
        }
```
and then change `td, th` to;
```
td, th {
            padding: 10px;
        }
```

See https://weasyprint.readthedocs.io/en/latest/tutorial.html#instantiating-html-and-css-objects

File: django_api/django_api/apps/unicef/exports/programme_documents.py (hunk 1 above)
Comment (reinbach) [reply django_api/django_api/apps/unicef/exports/programme_documents.py:223]: So the `render_pdf_to_response` function is ours in `apps/unicef/utils.py` that creates a FileResponse.
The function does a fair number of different operations, template render to string, write to pdf, write to tempfile, and create FileResponse, not sure there is a single exception to wrap that, so the generic Exception was used at this point

**Pull request merged**